functions.go

```diff
@@ -350,6 +351,28 @@ func Is(err, target error) bool {
 	return stderrors.Is(err, target)
 }
 
+// IsType is a convenience method for ascertaining if an error contains the
+// target error type within it's chain. This is aimed at ease of development
+// where a more complicated error type wants to be to checked for existence but
+// pointer var of that type is too much overhead.
+func IsType(err, target error) bool {
+	if target == nil {
+		panic("errors: target cannot be nil")
```

Comment:
My first reaction here was production code should not panic due to a programming error. We have had enough cases of juju agents panicing due to this sort of thing.
If target is nil we could just return false. But then we'd not pick up on the error. But that's better than unexpected panics. And it would follow what we do elsewhere with similar existing error helpers like `IsNotFound()` etc. So on balance I'd like to avoid the panic.